pkgs/by-name/pk/pkl/package.nix

```diff
@@ -8,11 +8,24 @@
   nix-update-script,
   replaceVars,
   makeWrapper,
+  graalvmPackages,
+  buildNative ? true,
 }:
 let
   jdk = temurin-bin-21;
   gradleOverlay = gradle.override { java = jdk; };
   kotlinOverlay = kotlin.override { jre = jdk; };
+  binaries = {
+    aarch64-darwin = "pkl-macos-aarch64";
+    aarch64-linux = "pkl-linux-aarch64";
+    x86_64-darwin = "pkl-macos-amd64";
+    x86_64-linux = "pkl-linux-amd64";
+    java = "jpkl";
+  };
+  inherit (stdenv.hostPlatform) system;
+  nativeBuild = (builtins.hasAttr system binaries) && buildNative;
```

Comment:
Good solution to automatically choose the java version when native isn't supported. I initially wanted to suggest a change to `meta.platforms` but this works even better!